gobblin-runtime/src/main/java/org/apache/gobblin/runtime/mapreduce/MRJobLauncher.java

```diff
@@ -228,8 +231,17 @@ public MRJobLauncher(Properties jobProps, Configuration conf, SharedResourcesBro
       this.fs.delete(this.mrJobDir, true);
     }
     this.unsharedJarsDir = new Path(this.mrJobDir, JARS_DIR_NAME);
-    this.jarsDir = this.jobProps.containsKey(ConfigurationKeys.MR_JARS_DIR) ? new Path(
-        this.jobProps.getProperty(ConfigurationKeys.MR_JARS_DIR)) : this.unsharedJarsDir;
+
+    if (this.jobProps.containsKey(ConfigurationKeys.MR_JARS_BASE_DIR)) {
+      Path jarsBaseDir = new Path(this.jobProps.getProperty(ConfigurationKeys.MR_JARS_BASE_DIR));
+      String monthSuffix = new SimpleDateFormat("yyyy-MM").format(System.currentTimeMillis());
+      cleanUpOldJarsDirIfRequired(this.fs, jarsBaseDir);
+      this.jarsDir = new Path(jarsBaseDir, monthSuffix);
+    } else {
```

Comment:
I think this would be easier having an else-if,
```
} else if (this.jobProps.containsKey(ConfigurationKeys.MR_JARS_DIR)) { 
  this.jarsDir = this.jobProps.getProperty(ConfigurationKeys.MR_JARS_DIR);
} else {
  this.jarsDir = this.unsharedJarsDir;
}
```

Reply:
That looks better, just wanted to make minimal changes to show in a diff to the reviewer :D